gym/envs/mujoco/mujoco_env.py

```diff
@@ -40,20 +58,26 @@ def __init__(self, model_path, frame_skip, rgb_rendering_tracking=True):
 
         self.init_qpos = self.sim.data.qpos.ravel().copy()
         self.init_qvel = self.sim.data.qvel.ravel().copy()
-        observation, _reward, done, _info = self.step(np.zeros(self.model.nu))
+
+        self._set_action_space()
+
+        action = self.action_space.sample()
+        observation, _reward, done, _info = self.step(action)
         assert not done
-        self.obs_dim = observation.size
 
+        self._set_observation_space(observation)
+
+        self.seed()
+
+    def _set_action_space(self):
```

Comment:
what is the purpose of putting this in a separate method vs keeping it in a constructor? Is it expected to be called or overridden in subclasses?

Reply:
Yeah, I think it would be nice to provide the ability for the subclasses to override the default behavior of these functions.